src/Router/UrlGenerator.php

```diff
@@ -47,40 +62,120 @@ public static function buildUrl($url, $replace = [], $flags = HTTP_URL_REPLACE,
                    | HTTP_URL_STRIP_PASS;
         }
 
-        // Filter $url and $replace arrays to strip out unknown segments
-        array_change_key_case($url, CASE_LOWER);
-        array_change_key_case($replace, CASE_LOWER);
+        // Parse input
+        if (is_string($url)) {
+            $url = parse_url($url);
+        }
+        if (is_string($replace)) {
+            $replace = parse_url($replace);
+        }
+
+        // Prepare input data
+        $cleanUrlArray = function (&$url) use ($urlSegments) {
+            if (!is_array($url)) {
+                $url = [];
+            }
+
+            foreach ($url as $key => &$value) {
+                // Remove invalid segments
```

Comment:
As far as I can tell, `$urlSegments` is not being used as a parameter and thus this segment removal is probably redundant. Segment removal seems to be happening on line 180 onwards.

Reply:
It's not a parameter, just a way to ensure only valid URL segments are in the $url and $replace properties. Line 180 has to do with stripping out valid but undesired segments.

Reply:
Gotcha